src/overlays/actors/ovl_Obj_Tokeidai/z_obj_tokeidai.c

```diff
@@ -50,87 +67,803 @@ const ActorInit Obj_Tokeidai_InitVars = {
+            ActorCutscene_GetCurrentIndex() == -1) {
+            this->actor.draw = NULL;
+        }
+        this->currentTime = gSaveContext.time;
+        if (this->actor.home.rot.x != 0) {
+            ObjTokeidai_Clock_Init(this);
+            this->actor.home.rot.x = 0;
+        }
+    }
+
+    if (gSaveContext.day % 5 != 3 || gSaveContext.time >= CLOCK_TIME(6, 0)) {
+        ObjTokeidai_RotateOnMinuteChange(this, true);
+    }
+    ObjTokeidai_RotateOnHourChange(this, globalCtx);
+}
+
+void ObjTokeidai_WallClock_Idle(ObjTokeidai* this, GlobalContext* globalCtx) {
+    this->currentTime = gSaveContext.time;
+    ObjTokeidai_RotateOnMinuteChange(this, true);
+    ObjTokeidai_RotateOnHourChange(this, globalCtx);
+}
+
+void ObjTokeidai_TowerGear_Idle(ObjTokeidai* this, GlobalContext* globalCtx) {
+    if (ObjTokeidai_IsPostFirstCycleFinalHours(this, globalCtx)) {
+        this->actor.draw = ObjTokeidai_TowerGear_Draw;
+    } else {
+        if (globalCtx->csCtx.state != 0) {
+            this->actor.home.rot.x = 1;
+            this->currentTime += 3;
+            this->actor.draw = ObjTokeidai_TowerGear_Draw;
+        } else {
+            if ((globalCtx->actorCtx.unk5 & 2) == 0 &&
+                OBJ_TOKEIDAI_TYPE(&this->actor) == OBJ_TOKEIDAI_TYPE_TOWER_GEAR_TERMINA_FIELD &&
+                ActorCutscene_GetCurrentIndex() == -1) {
+                this->actor.draw = NULL;
+            }
+            this->currentTime = gSaveContext.time;
+            if (this->actor.home.rot.x != 0) {
+                ObjTokeidai_SetupClockOrGear(this);
+                this->actor.home.rot.x = 0;
+            }
+        }
+        ObjTokeidai_RotateOnMinuteChange(this, false);
+    }
+}
+
+void ObjTokeidai_Counterweight_Idle(ObjTokeidai* this, GlobalContext* globalCtx) {
+    s32 type;
+
+    if (ObjTokeidai_IsPostFirstCycleFinalHours(this, globalCtx)) {
+        this->spotlightIntensity = 0;
+        if ((this->actor.child == NULL)) {
+            type = OBJ_TOKEIDAI_TYPE(&this->actor);
+            if (type == OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_CLOCK_TOWN ||
+                type == OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_TERMINA_FIELD) {
+                this->actor.child =
+                    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_HANABI, this->actor.world.pos.x,
+                                this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
+            }
+        }
+        if (this->actor.child != NULL) {
+            if (OBJ_TOKEIDAI_TYPE(&this->actor) == OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_TERMINA_FIELD) {
+                this->actor.child->home.rot.x = 0x384;
+            } else {
+                this->actor.child->home.rot.x = 0x12C;
+            }
+        }
+    } else {
+        this->actor.shape.rot.y -= 0x40;
+        if (gSaveContext.isNight != 0) {
+            if (this->spotlightIntensity < 100) {
+                this->spotlightIntensity += +4;
```

Comment:
```suggestion
                this->spotlightIntensity += 4;
```

Reply:
How in the world did I miss this one? Thanks for catching it.